### PR TITLE
Use recording rule names instead of label selectors

### PR DESCRIPTION
**Pull request by krasi-georgiev**
Signed-off-by: Krasi Georgiev <8903888+krasi-georgiev@users.noreply.github.com>

### CODE REVIEW

**Comment by metalmatze** [comment]
Hmm, honestly. A recording rule like `status_class_5xx:rate_1h_by_code:promhttp_metric_handler_requests_total:namespace:default:job:fooapp ` is going too far off from the best practices we have around recording rule names.

**Comment by metalmatze** [comment]
I thought of this more in the realms of being able to specificy a intermediate recording rule name rather than adding even more to the already existing one.

**Comment by metalmatze** [comment]
Essentially some way of setting the intermediate recording rule name to
```diff
-promhttp_metric_handler_requests_total
+promhttp_metric_handler_requests_ratio
```
and thus the recording rule itself could become
```diff
-status_class_5xx:promhttp_metric_handler_requests_total:ratio_rate1h{namespace="default",job="fooapp"}
+status_class_5xx:promhttp_metric_handler_requests_ratio:rate1h{namespace="default",job="fooapp"}

```

**Comment by metalmatze** [comment]
Basically adding a field like 
`recordingMetric` next to 
https://github.com/metalmatze/slo-libsonnet/blob/9baab703bb8a55f990f3b6a74c80987036938f64/slo-libsonnet/error-burn.libsonnet#L7

File: examples/http-request-errors.yaml [hunk 1]
```diff
@@ -1,32 +1,28 @@
...
     ) > 0.050000000000000003
   for: 5m
   labels:
-    job: fooapp
-    namespace: default
     severity: warning
 - expr: |
     (
-      sum(status_class:promhttp_metric_handler_requests_total:rate5m{status_class="5xx"})
+      sum(rate_5m_by_code:promhttp_metric_handler_requests_total{status_class="5xx"})
```
Comment (krasi-georgiev) [inline examples/http-request-errors.yaml:13]: hm I am not sure about this one. Does the recording rule add this `status_class` label?

**Comment by krasi-georgiev** [comment]
surpresed by https://github.com/metalmatze/slo-libsonnet/pull/40/files

**Pull request closed without merging**